### PR TITLE
Update dependencies to latest version and fix overlay dialog.

### PR DESCRIPTION
**Pull request by epool**
- Update target version from api `23` to api `25`.
- Update project dependencies to latest versions.
- FIX: Overlay Controller allows to click and scroll parent controller. [DEMO]

### CODE REVIEW

File: dependencies.gradle [hunk 1]
```diff
@@ -1,30 +1,30 @@
 ext {
     minSdkVersion = 16
-    compileSdkVersion = 23
-    targetSdkVersion = 23
-    buildToolsVersion = '23.0.2'
+    compileSdkVersion = 24
+    targetSdkVersion = 24
+    buildToolsVersion = '24.0.1'
```
Comment (PaulWoitaschek) [inline dependencies.gradle]: Latest build tools is 24.0.3

**Comment by PaulWoitaschek** [comment]
Also you should change the [.travis.yml](https://github.com/bluelinelabs/Conductor/blob/develop/.travis.yml) file to match with the versions

File: dependencies.gradle [hunk 2]
```diff
@@ -1,30 +1,30 @@
...
-    buildToolsVersion = '23.0.2'
+    compileSdkVersion = 24
+    targetSdkVersion = 24
+    buildToolsVersion = '24.0.3'
 
-    supportV4 = 'com.android.support:support-v4:23.1.1'
-    supportDesign = 'com.android.support:design:23.1.1'
-    supportAnnotations = 'com.android.support:support-annotations:23.1.1'
-    supportAppCompat = 'com.android.support:appcompat-v7:23.1.1'
+    supportV4 = 'com.android.support:support-v4:24.1.1'
```
Comment (PaulWoitaschek) [inline dependencies.gradle]: Also latest support library is 
24.2.1

File: dependencies.gradle (hunk 1 above)
Comment (epool) [inline dependencies.gradle]: done!

**Comment by PaulWoitaschek** [comment]
👍

**Comment by dimsuz** [comment]
I would strongly suggest to manually test that everything using support libraries (I assume demo application mainly) is working as intended after this change. Almost every update of appcompact/design brought in some regressions or new bugs in my experience...

**Comment by epool** [comment]
@dimsuz I have already tested the demo project manually on my Nexus 5x. Everything looks to be working well.

File: dependencies.gradle [hunk 3]
```diff
@@ -1,30 +1,30 @@
...
 
-    leakCanary = 'com.squareup.leakcanary:leakcanary-android:1.4'
-    leakCanaryNoOp = 'com.squareup.leakcanary:leakcanary-android-no-op:1.4'
+    leakCanary = 'com.squareup.leakcanary:leakcanary-android:1.5'
+    leakCanaryNoOp = 'com.squareup.leakcanary:leakcanary-android-no-op:1.5'
 
-    rxJava = 'io.reactivex:rxjava:1.2.0'
+    rxJava = 'io.reactivex:rxjava:1.2.1'
     rxAndroid = 'io.reactivex:rxandroid:1.2.1'
     rxLifecycle = 'com.trello:rxlifecycle:0.8.0'
```
Comment (saantiaguilera) [inline dependencies.gradle]: Is it possible to update this to `1.0` :D

**Pull request merged**